packages/@tailwindcss-vite/src/index.ts

```diff
@@ -83,26 +83,20 @@ export default function tailwindcss(): Plugin[] {
     }
 
     if (updated) {
-      invalidateAllRoots(isSSR)
+      invalidateAllRoots()
     }
   }
 
-  function invalidateAllRoots(isSSR: boolean) {
+  function invalidateAllRoots() {
+    let rootsFound = new Set()
+
     for (let server of servers) {
       let updates: Update[] = []
-      for (let [id, root] of roots.entries()) {
+      for (let [id] of roots.entries()) {
         let module = server.moduleGraph.getModuleById(id)
-        if (!module) {
-          // Note: Removing this during SSR is not safe and will produce
-          // inconsistent results based on the timing of the removal and
-          // the order / timing of transforms.
-          if (!isSSR) {
```

Comment:
I'm not 100% sure why this is safe to do this unconditionally. Is it just because we are never removing any `root`s anymore?

Reply:
We're no longer calling `roots.delete(id)` so this check can be removed too, not sure I understand your question correctly tho? 😅

Reply:
I'm being a dumb dumb, carry on.